Assembless/react-littera
# Implement Trans component

## 📜 Description

This PR adds a new feature to the project: the Trans component. The Trans component is a custom translation component that can interpolate variables and parse HTML elements.

The Trans component takes three props:

- `children` - The translation string to be rendered.
- `values` - An object containing values to be interpolated in the translation string.
- `components` - An object containing custom React components or HTML tags to be parsed and rendered.

This PR also includes an updated version of `package.json`.

## 🧪 How to test this?

To test this PR, follow these steps:

1. Clone the repository.
2. Checkout the `trans-component` branch.
3. Install dependencies with `yarn install`.
4. Run tests with `yarn test`.
5. Link littera package with `yarn link` while in the root directory.
6. Set up an example React app with `npx create-react-app ./PlaygroundApp --template=typescript`
7. Link littera with the example app and complete basic setup. Link with `yarn link @assembless/react-littera` while in the apps root directory.
6. Run the app and navigate to `http://localhost:3000` to see the Trans component in action.

## 🚀 Examples

Here are a few examples of how to use the Trans component:

```jsx
import React from 'react';
import { Trans, useLittera } from '@assembless/react-littera';

const translations = {
  welcome: {
    en_US: "Welcome <strong>{name}</strong>!",
    de_DE: "Willkommen <strong>{name}</strong>!",
    pl_PL: "Witaj <strong>{name}</strong>!",
  },
};

function App() {
  const translated = useLittera(translations);

  return (
    <Trans values={{ name: "Jack" }}>
      {translated.welcome}
    </Trans>
  );
}
```

```jsx
import React from 'react';
import { Trans, useLittera } from '@assembless/react-littera';

const translations = {
  about: {
    en_US: "This is the <a href='/about'>about</a> page.",
    de_DE: "Dies ist die <a href='/about'>Über uns</a>-Seite.",
    pl_PL: "To jest <a href='/about'>strona o nas</a>.",
  },
};

function AboutPage() {
  const translated = useLittera(translations);

  return (
    <Trans components={{ a: "Link" }}>
      {translated.about}
    </Trans>
  );
}
```
